File: CSES_ALL/1.py
s = input().strip()

n = len(s)
prefix_sum = [0] * (n + 1)

for i in range(n):
    prefix_sum[i + 1] = prefix_sum[i] + int(s[i])

max_len = 0

for length in range(2, n + 1, 2):
    for i in range(n - length + 1):
        mid = i + length // 2
        left_sum = prefix_sum[mid] - prefix_sum[i]
        right_sum = prefix_sum[i + length] - prefix_sum[mid]

        if left_sum == right_sum:
            max_len = max(max_len, length)

print(max_len)

File: CSES_ALL/1.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    string s;
    cin >> s;

    int n = s.length();
    vector<int> prefix_sum(n + 1, 0);

    for (int i = 0; i < n; i++) {
        prefix_sum[i + 1] = prefix_sum[i] + (s[i] - '0');
    }

    int max_len = 0;

    for (int len = 2; len <= n; len += 2) {
        for (int i = 0; i + len <= n; i++) {
            int mid = i + len / 2;
            int left_sum = prefix_sum[mid] - prefix_sum[i];
            int right_sum = prefix_sum[i + len] - prefix_sum[mid];

            if (left_sum == right_sum) {
                max_len = max(max_len, len);
            }
        }
    }

    cout << max_len << endl;

    return 0;
}
